ichnaea/async/tests.py

```diff
@@ -10,38 +9,85 @@
-
-        assert (all_tasks - registered_tasks ==
-                set(['data.update_blue', 'data.update_cell',
-                     'data.update_wifi',
-                     'data.update_datamap', 'data.cleanup_datamap',
-                     'data.cell_export_diff', 'data.cell_export_full',
-                     'data.export_reports', 'data.update_statregion',
-                     ]))
-
-        assert (set(['_'.join(name.split('_')[:-1]) for name in
-                     registered_tasks - all_tasks]) ==
-                set(['data.update_blue', 'data.update_cell',
-                     'data.update_wifi']))
+        all_tasks = set([
+            m[1].shortname() for m in getmembers(tasks)
+            if isinstance(m[1], BaseTask)
+        ])
+
+        assert (
+            all_tasks - registered_tasks
+            == set([
+                'data.update_blue',
+                'data.update_cell',
+                'data.update_wifi',
+                'data.update_datamap',
+                'data.cleanup_datamap',
+                'data.cell_export_diff',
+                'data.cell_export_full',
+                'data.export_reports',
+            ])
+        )
+
+        assert (
+            registered_tasks - all_tasks
+            == set([
+                'data.cleanup_datamap_ne',
+                'data.cleanup_datamap_nw',
+                'data.cleanup_datamap_se',
+                'data.cleanup_datamap_sw',
+                'data.update_blue_0',
+                'data.update_blue_1',
+                'data.update_blue_2',
+                'data.update_blue_3',
+                'data.update_blue_4',
+                'data.update_blue_5',
+                'data.update_blue_6',
+                'data.update_blue_7',
+                'data.update_blue_8',
+                'data.update_blue_9',
+                'data.update_blue_a',
+                'data.update_blue_b',
+                'data.update_blue_c',
+                'data.update_blue_d',
+                'data.update_blue_e',
+                'data.update_blue_f',
+                'data.update_cell_gsm',
+                'data.update_cell_lte',
+                'data.update_cell_wcdma',
+                'data.update_datamap_ne',
+                'data.update_datamap_nw',
+                'data.update_datamap_se',
+                'data.update_datamap_sw',
+                'data.update_wifi_0',
+                'data.update_wifi_1',
+                'data.update_wifi_2',
+                'data.update_wifi_3',
+                'data.update_wifi_4',
+                'data.update_wifi_5',
+                'data.update_wifi_6',
+                'data.update_wifi_7',
+                'data.update_wifi_8',
+                'data.update_wifi_9',
+                'data.update_wifi_a',
+                'data.update_wifi_b',
+                'data.update_wifi_c',
+                'data.update_wifi_d',
+                'data.update_wifi_e',
+                'data.update_wifi_f',
```

Comment:
Before, `registered_tasks` would group the tasks dropping the right-most part. That seems curious to me--seems like we really want to compare against the complete list of tasks.

Reply:
My guess is that this was to minimize the test changes when these tasks exploded into buckets. I like the more explicit list.

Reply:
That's interesting. I bet you're right.